### ISSUE TITLE
Test that the GPC headers & other signals clear in time after running `Analysis` mode

### ISSUE THREAD

**Issue by kalicki1** [privacy-tech-lab/gpc-optmeowt#234]
Does our extension send "remnant" GPC signals to a freshly navigated site after running an analysis on the previous site? In other words: 

(1) I navigate to `chipotle.com`, then (2) I send GPC signals to `chipotle.com`, force it to reload, and collect information about it as part of our analysis, and then (3) immediately navigate to `shakeshack.com`.

Does `shakeshack.com` have any GPC signals attached to it? Specifically, will there be rogue GPC headers in some requests?

**Comment by SebastianZimmeck** [comment]
Per our discussion today, it is not clear what this issue is about. There was some situation when an analysis of a website was run multiple times and some results where not properly stored. @kalicki1 will try to replicate the behavior of analyzing a site multiple times and check whether analysis results are stored correctly. If that is the case, close the issue. Otherwise, find out where the problem comes from and fix it.

**Comment by kalicki1** [comment]
I have been trying to remember what this issue could be about and it struck me while I was working on something else:

I wanted to know that our extension wasn't sending GPC signals to any site we navigate to immediately after running the analysis mode on one site. For example, say (1) I navigate to `chipotle.com`, then (2) I send GPC signals to `chipotle.com`, force it to reload, and collect information about it as part of our analysis, and then (3) immediately navigate to `shakeshack.com`.

The point is that `shakeshack.com` shouldn't have any GPC signals attached to it. 

**Do we actually do the correct thing here?**

I tested this on our [GPC reference implementation](https://global-privacy-control.glitch.me/) and only the very first request receives the GPC header, though it's still not ideal (no DOM property is set). This most likely has to do with how I determine when a new site is being navigated to with the `webNavigation.onCommitted` listener which does not map well to exact requests. 

I will push a change when I can figure out how to patch this.

**Comment by kalicki1** [comment]
I think a half-way measure could be to check the `documentUrl` of all the incoming requests associated with a tab against what was known to be be the previous top level `documentUrl`. 

This should catch navigations away from a site, say from `example.com` to `sample.net`, more accurately than the current method. It could still be iffy for reloads or navigations into different pages on the same domain, such as going from an `example.com` site to another `example.com` site, but I will test this out.

**Comment by kalicki1** [comment]
Per what was discussed in our last meeting with @SebastianZimmeck, I decided to implement a manual, analysis mode shut-off that looks like this:
 
<img width="331" alt="Screen Shot 2021-11-24 at 4 51 48 AM" src="https://user-images.githubusercontent.com/60238705/143215421-b24dc9f0-c08f-4440-baa6-e04dea822201.png">

I tried to fix this issue without a manual stop, but this resolved the issue very nicely and I decided to keep it. I used this implementation in my test mentioned in #238. 

Since this is all set, I will go ahead and close this issue.
